pkg/discovery/service.go

```diff
@@ -39,32 +41,57 @@ func NewDiscovery(logger *zap.Logger, ol *overlay.Cache, kad *kademlia.Kademlia,
 	}
 }
 
+// StartRefresh kicks off a goroutine that refreshes the cache on an interval and returns an error
+func (d *Discovery) StartRefresh(ctx context.Context) error {
+	var errs []error
+	go func() {
+		t := time.NewTicker(500 * time.Millisecond).C
+		for {
+			<-t
+			if err := d.Refresh(ctx); err != nil {
+				d.log.Error(err.Error())
+				errs = append(errs, err)
+			}
+		}
+	}()
+	return utils.CombineErrors(errs...)
```

Comment:
the for loop can never exit, an hence this never gets called... we're just making a potentially infinite list of errors in the meantime